candidate_models/model_commitments/cornets.py

```diff
@@ -356,20 +355,20 @@ def __init__(self):
 
         for basemodel_identifier, brain_model in model_pool.items():
             activations_model = LazyLoad(lambda brain_model=brain_model: brain_model.activations_model)
-            for identifier, activations_model in Hooks().iterate_hooks(basemodel_identifier, activations_model):
-                def load(basemodel_identifier=basemodel_identifier, identifier=identifier, brain_model=brain_model):
-                    # only update when actually required, otherwise we'd change the activations_model
-                    # of one brain_model at all times
-                    if basemodel_identifier in self._accessed_brain_models:
-                        raise ValueError(f"{identifier}'s brain-model {basemodel_identifier} has already been accessed "
-                                         f"in this session. To avoid clashes in the hooks, "
-                                         f"please run {identifier} in a separate session.")
-                    self._accessed_brain_models.append(basemodel_identifier)
-                    # upon accessing the `activations_model`, the Hook will automatically
-                    # attach to the `brain_model.activations_model`.
-                    return brain_model
-
-                self[identifier] = LazyLoad(load)
+            # for identifier, activations_model in Hooks().iterate_hooks(basemodel_identifier, activations_model):
+            def load(basemodel_identifier=basemodel_identifier, identifier=basemodel_identifier, brain_model=brain_model):
+                # only update when actually required, otherwise we'd change the activations_model
+                # of one brain_model at all times
+                if basemodel_identifier in self._accessed_brain_models:
+                    raise ValueError(f"{identifier}'s brain-model {basemodel_identifier} has already been accessed "
+                                     f"in this session. To avoid clashes in the hooks, "
+                                     f"please run {identifier} in a separate session.")
```

Comment:
this should follow the other retrieval now, right? i.e. allow for multiple retrieval, but warn about hook unregistering

Reply:
still needs addressing